hoodie-bench/src/main/java/com/uber/hoodie/integrationsuite/generator/WorkloadGenerator.java

```diff
@@ -0,0 +1,196 @@
+/*
+ *  Copyright (c) 2019 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+package com.uber.hoodie.integrationsuite.generator;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.uber.hoodie.integrationsuite.configuration.WorkloadOperationConfig;
+import com.uber.hoodie.integrationsuite.converter.InsertToUpdateConverter;
+import com.uber.hoodie.integrationsuite.sink.SinkType;
+import com.uber.hoodie.integrationsuite.sink.WorkloadSink;
+import com.uber.hoodie.integrationsuite.sink.WorkloadSinkFactory;
+import com.uber.hoodie.integrationsuite.sink.config.DFSSinkConfig;
+import com.uber.hoodie.integrationsuite.sink.config.SinkConfig;
+import com.uber.hoodie.integrationsuite.sink.reader.DFSAvroSinkReader;
+import com.uber.hoodie.integrationsuite.sink.writer.WriteStats;
+import com.uber.hoodie.utilities.sources.converter.Converter;
+import java.io.IOException;
+import java.io.Serializable;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.StreamSupport;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.storage.StorageLevel;
+import scala.Tuple2;
+
+/**
+ * The workload generator generates all types of workloads (insert, update) for the given configs
+ */
+public class WorkloadGenerator implements Serializable {
```

Comment:
Rename to DeltaGenerator/DeltaProducer? Seems we are overloading the term Workload? Also plays nicely with DeltaStreamer in the sense, we generate deltas and then the deltastreamer consumes them